### ISSUE TITLE
Make use of internal sensors when sky is not clearly visible

### ISSUE THREAD

**Issue by serrq** [organicmaps/organicmaps#6048]
When trees and buildings obscure the sky I lose my navigation while I am driving.

I would that app automatically calculate route with internal sensors until the sky come visible again.

**Comment by janusbenissa** [comment]
What  internal sensors are you meaning?

**Comment by janusbenissa** [comment]
If you don't have satellite repression, your phone don't know your speed, so it will be extreme difficult to calculate your position.

**Comment by RicoElectrico** [comment]
What you're asking for is called _inertial navigation_ or _dead reckoning_ and it's pretty tough to solve (google it). To some extent, I think built-in car navigations use it by the means of wheel speed (and other) sensors.

**Comment by RicoElectrico** [comment]
> The ICBM intercontinental missile has this system as backup when flying... and for some reason gps is not reached... It hasn't wheels.

Yeah, it has a crazy expensive gyroscope.

**Comment by Misalf-git** [comment]
That'd be fascinating.
While my phone is in continuous movement, how would acceleration be recognized? Why am I not falling off the planet?

**Comment by Misalf-git** [comment]
Ah, so continuous **_change_** in movement is required to get any readings.
I'll make the wild guess that this wouldn't result in meaningful estimates very often.

**Comment by auge8472** [comment]
> Ah, so continuous ***change*** in movement is required to get any readings.

Yeah, the accelerometer measures acceleration (positive (speeding up) and negative (braking)). 
😀

**Comment by muralito** [comment]
Isn't the inertial navigation also needed for indoor navigation? like when you are in a shopping mall or public transport building?

**Comment by muralito** [comment]
There seems not easy to implement. https://stackoverflow.com/questions/7829097/android-accelerometer-accuracy-inertial-navigation

BTW, sometimes I use this, https://f-droid.org/es/packages/org.secuso.privacyfriendlyactivitytracker/ and is not perfect but it's accuracy is enough for me, but one thing is count, and other estimate location with inertial navigation.

**Comment by uis246** [comment]
> > The ICBM intercontinental missile has this system as backup when flying... and for some reason gps is not reached... It hasn't wheels.
> 
> Yeah, it has a crazy expensive gyroscope.

And phones have crazy inexpensive gyroscopes. And accelerometers.

**Comment by uis246** [comment]
1. This is huge
2. Using algorithms from opensource firmware of some flight computer might help, but this is still very hard task
